tourney/main.py

```diff
@@ -70,22 +71,67 @@ def create_teams():
 
   return teams, names
 
-def pick_pairs(amount):
-  """Picks non-overlapping team pairs of 2 rounds."""
-  return [(i, i + 1, 2) for i in range(0, amount, 2)]
-
-def create_schedule(amount):
-  """Takes amount of teams to schedule for."""
+def all_team_combinations(teams):
+  """Returns all combinations of pairs of teams"""
+  if len(teams) < 2:
+    yield []
+    return
+  elif len(teams) == 3:
+    # Only one combination possible here
+    yield [(teams[0], teams[1], 1),
+           (teams[1], teams[2], 1),
+           (teams[0], teams[2], 1)]
+    return
+  elif len(teams) % 2 == 1:
+    # Handle odd length list. Make one XvXvX and (n-3)/2 pairs.
+    # Combine each possible triple with each possible configuration of the other teams
+    for i in range(len(teams)):
+      i_remains = list(range(len(teams)))
+      i_remains.remove(i)
+      for j in i_remains:
+        j_remains = [n for n in i_remains]
+        j_remains.remove(j)
+        for k in j_remains:
+          triple = [(teams[i], teams[j], 1), (teams[i], teams[k], 1), (teams[j], teams[k], 1)]
+          remains = [t for t in teams if t not in [teams[i], teams[j], teams[k]]]
+          for result in all_team_combinations(remains):
+            yield triple + result
+  else:
+    a = teams[0]
+    for i in range(1, len(teams)):
+      # Pick a pair and recurse on rest of list
+      pair = (a, teams[i], 2)
+      for r in all_team_combinations(teams[1:i] + teams[i + 1:]):
+        yield [pair] + r
+
+def create_schedule(teams):
+  """Takes list of teams to schedule for."""
   matches = []
-  if amount % 2 == 0:
-    matches = pick_pairs(amount)
+  if MATCHMAKING:
+    player_skill = PlayerSkill.get()
+    all_combinations = list(all_team_combinations(list(range(len(teams)))))
```

Comment:
`len(teams)` is used a lot in this function. Maybe use as `amount = len(teams)` instead.

Reply:
Same in `all_team_combinations()` above.